app/src/main/java/fr/neamar/kiss/searcher/Searcher.java

```diff
@@ -2,52 +2,180 @@
 
 
 import android.os.AsyncTask;
+import android.os.Handler;
+import android.os.Looper;
+import android.support.annotation.CallSuper;
 import android.util.Log;
 
+import java.lang.ref.WeakReference;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
-import java.util.List;
+import java.util.PriorityQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 
 import fr.neamar.kiss.MainActivity;
 import fr.neamar.kiss.pojo.Pojo;
+import fr.neamar.kiss.pojo.PojoComparator;
 import fr.neamar.kiss.result.Result;
 
-public abstract class Searcher extends AsyncTask<Void, Void, List<Pojo>> {
-    static final int DEFAULT_MAX_RESULTS = 25;
-    private long start;
-    private String query;
+public abstract class Searcher extends AsyncTask<Void, Result, Void>
+{
+	static final        int             DEFAULT_MAX_RESULTS   = 50;
+	static final        int             DEFAULT_REFRESH_TIMER = 150;
+	// define a different thread than the default AsyncTask thread or else we will block everything else that uses AsyncTask while we search
+	public static final ExecutorService SEARCH_THREAD         = Executors.newSingleThreadExecutor();
```

Comment:
I think you use the same thread to display images too, would it make more sense to use one thread for searching and one for displaying images?

Reply:
I do not. I made the SEARCH_THREAD because I wanted the searching to be done on a different thread. Image loading is using execute() and searching is using executeOnExecutor...